src/kOS/AddOns/Trajectories/TRWrapper.cs

```diff
@@ -105,8 +111,10 @@ private static void init()
             return (Vector3?)trPlannedDirection.Invoke(null, new object[] { });
         }
 
-        public static bool HasTarget()
+        public static bool? HasTarget()
         {
+            if (trHasTarget == null)
+                return null;
             return (bool)trHasTarget.Invoke(null, new object[] { });
```

Comment:
Weirdly, I tried to do the above using the C# shorthand:
``(bool?)trHasTarget?.Invoke(null, new object[] { });``
and the compiler complained.  I'm not sure why, as we are using this syntax feature elsewhere.

Reply:
That will because I have not set `HasTarget` as a nullable type.
`public static bool HasTarget()`

 I will update it to:
`public static bool? HasTarget()`

Reply:
actually you have used a cast, so it will make no difference if I change the API, maybe bool is not nullable !

Reply:
I'm using VS2017 and compilation went fine. build also works fine testing on my system.